src/coreComponents/constitutive/fluid/MultiFluidBase.cpp

```diff
@@ -193,8 +203,52 @@ void MultiFluidBase::initializeState( arrayView2d< real64 const, compflow::USD_P
       }
     }
   } );
+
+  // initialize the "old" variables
+  saveConvergedState();
+}
+
+void MultiFluidBase::saveConvergedState() const
+{
+  localIndex const numElem = m_initialTotalMassDensity.size( 0 );
+  localIndex const numGauss = m_initialTotalMassDensity.size( 1 );
+  integer const numPhase = m_phaseMassDensity.value.size( 2 );
+  integer const numComp = m_phaseCompFraction.value.size( 3 );
+
+  FluidProp::ViewTypeConst const totalDensity = m_totalDensity.toViewConst();
+  PhaseProp::ViewTypeConst const phaseDensity = m_phaseDensity.toViewConst();
+  PhaseProp::ViewTypeConst const phaseEnthalpy = m_phaseEnthalpy.toViewConst();
+  PhaseProp::ViewTypeConst const phaseInternalEnergy = m_phaseInternalEnergy.toViewConst();
+  PhaseComp::ViewTypeConst const phaseCompFraction = m_phaseCompFraction.toViewConst();
+
+  arrayView2d< real64, multifluid::USD_FLUID > const totalDensityOld = m_totalDensityOld.toView();
+  arrayView3d< real64, multifluid::USD_PHASE > const phaseDensityOld = m_phaseDensityOld.toView();
+  arrayView3d< real64, multifluid::USD_PHASE > const phaseEnthalpyOld = m_phaseEnthalpyOld.toView();
+  arrayView3d< real64, multifluid::USD_PHASE > const phaseInternalEnergyOld = m_phaseInternalEnergyOld.toView();
+  arrayView4d< real64, multifluid::USD_PHASE_COMP > const phaseCompFractionOld = m_phaseCompFractionOld.toView();
+
+  forAll< parallelDevicePolicy<> >( numElem, [=] GEOSX_HOST_DEVICE ( localIndex const k )
+  {
+    for( localIndex q = 0; q < numGauss; ++q )
+    {
+      totalDensityOld[k][q] = totalDensity.value[k][q];
+      for( integer ip = 0; ip < numPhase; ++ip )
+      {
+        phaseDensityOld[k][q][ip] = phaseDensity.value[k][q][ip];
+        phaseEnthalpyOld[k][q][ip] = phaseEnthalpy.value[k][q][ip];
+        phaseInternalEnergyOld[k][q][ip] = phaseInternalEnergy.value[k][q][ip];
+        for( integer ic = 0; ic < numComp; ++ic )
+        {
+          phaseCompFractionOld[k][q][ip][ic] = phaseCompFraction.value[k][q][ip][ic];
+        }
```

Comment:
So we do not store the derivatives, right?

Reply:
well, for the quantities at the previous time-step we only need the value non its derivatives.